OSX main menu bar

TheSpiritXIII opened the pull request:
Adds a global menu bar for OS X instead of having an empty one. Includes common OS X shortcuts/menus, including preferences, about and quit, as well as a few window-based shortcuts.

This does not include some actions such as cut/copy/paste because they're a little bit harder to implement for OS X. It would require signals on each widget which is tedious work and will likely be a different PR.

TheSpiritXIII committed:
OSX main menu bar

tux3 commented:
Your PR and commit message don't match what you changed. Don't do that.

It looks like you tried to fix the INSTALL.md, that's good, but put that change in a separate and properly named commit. I'm sure that was a mistake, but be careful please. Don't just hide unrelated changes in.

TheSpiritXIII commented:
The change you're talking about is what I told you about before where you merged my OS X install instructions too early. zetok told me to fix the readme format for OS X in the IRC channel to match the others. I fixed it and squashed it but I was a few minutes too late and it could no longer be merged properly.

tux3 commented:
@zetok stop telling people to squash things, it messes with the history.

TheSpiritXIII commented:
Removed the INSTALL.md changes.

TheSpiritXIII commented:
Don't merge yet. Just realized that the menu bar doesn't play nicely with the login screen (since it's a global menu bar).

EDIT: It's working now.

TheSpiritXIII committed:
OSX usable menu bar on login screen

tux3 commented on src/widget/widget.cpp:
```diff
@@ -342,6 +415,11 @@ Widget::~Widget()
+    //Nexus::getInstance().globalMenuBar->clear();
+    //Nexus::getInstance().globalMenuBar->addMenu(Nexus::getInstance().windowMenu);
+#endif
 }
```
Can be removed?

TheSpiritXIII committed:
OSX: Remove dead code and make Widget own all their actions
